src/foam/nanos/dig/DUGDAO.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.dig;
+
+import foam.core.FObject;
+import foam.core.X;
+import foam.dao.JDAO;
+import foam.dao.ProxyDAO;
+import foam.nanos.auth.User;
+
+public class DUGDAO
+    extends ProxyDAO
+{
+  public DUGDAO(X x) {
+    setX(x);
+    setDelegate(new JDAO(x, DUG.getOwnClassInfo(), "dugs"));
```

Comment:
This design doesn't re-execute() DUG's on startup. It would have to do a select() in the constructor to find all existing DUG's to start or don't setDelegate() and insert between the JDAO and the MapDAO so that all values automatically get re-put_() on startup. Second approach isn't as general though, because it wouldn't work with a Postgres DAO for example.

Reply:
Modified to select all DUGS on start